packages/scaffold-ui/test/views/w3m-profile-wallets-view.test.ts

```diff
@@ -0,0 +1,884 @@
+} as ConnectorWithProviders
+
+const mockEthereumAddress = 'eip155:1:0x1234567890123456789012345678901234567890'
+const mockBitcoinAddress =
+  'bip122:000000000019d6689c085ae165831e93:bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh'
+
+const mockActiveConnection: Connection = {
+  connectorId: 'metamask',
+  accounts: [
+    { address: '0x1234567890123456789012345678901234567890', type: 'eoa' },
+    { address: '0x9876543210987654321098765432109876543210', type: 'smart' }
+  ]
+}
+
+const mockRecentConnection: Connection = {
+  connectorId: 'walletconnect',
+  accounts: [{ address: '0xabcdefabcdefabcdefabcdefabcdefabcdefabcd', type: 'eoa' }]
+}
+
+const mockBitcoinConnection: Connection = {
+  connectorId: 'bitcoin-wallet',
+  accounts: [
+    { address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh', type: 'payment' },
+    { address: 'tb1qw508d6qejxtdg4y5r3zarvary0c5xw7kxpjzsx', type: 'ordinals' }
+  ]
+}
+
+// Mock ResizeObserver
+beforeAll(() => {
+  global.ResizeObserver = class {
+    observe() {}
+    unobserve() {}
+    disconnect() {}
+  }
+})
+
+describe('W3mProfileWalletsView - Basic Rendering', () => {
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    // Setup default state
+    vi.spyOn(ChainController, 'state', 'get').mockReturnValue({
+      activeChain: ConstantsUtil.CHAIN.EVM,
+      activeCaipNetwork: mockEthereumNetwork,
+      chains: new Map([
+        [
+          ConstantsUtil.CHAIN.EVM,
+          { namespace: ConstantsUtil.CHAIN.EVM, caipNetworks: [mockEthereumNetwork] }
+        ],
+        [
+          ConstantsUtil.CHAIN.SOLANA,
+          { namespace: ConstantsUtil.CHAIN.SOLANA, caipNetworks: [mockSolanaNetwork] }
+        ],
+        [
+          ConstantsUtil.CHAIN.BITCOIN,
+          { namespace: ConstantsUtil.CHAIN.BITCOIN, caipNetworks: [mockBitcoinNetwork] }
+        ]
+      ])
+    } as unknown as ChainControllerState)
+
+    vi.spyOn(ConnectorController, 'state', 'get').mockReturnValue({
+      ...ConnectorController.state,
+      activeConnectorIds: {
+        [ConstantsUtil.CHAIN.EVM]: 'metamask'
+      } as unknown as Record<ChainNamespace, string | undefined>,
+      connectors: [mockMetaMaskConnector, mockWalletConnectConnector, mockAuthConnector]
+    })
+
+    vi.spyOn(ChainController, 'getAccountData').mockReturnValue({
+      caipAddress: mockEthereumAddress as CaipAddress,
+      profileName: 'Test Profile'
+    } as unknown as AccountControllerState)
+
+    vi.spyOn(ConnectionControllerUtil, 'getConnectionsData').mockReturnValue({
+      connections: [mockActiveConnection],
+      recentConnections: [mockRecentConnection]
+    })
+
+    vi.spyOn(ConnectionController, 'subscribeKey').mockReturnValue(() => {})
+    vi.spyOn(ConnectorController, 'subscribeKey').mockReturnValue(() => {})
+    vi.spyOn(ChainController, 'subscribeKey').mockReturnValue(() => {})
+    vi.spyOn(ChainController, 'subscribeChainProp').mockReturnValue(() => {})
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
```

Comment:
We already have this on beforeEach, do we still need? If not let's remove

Reply:
No, I think this is ok. This will clear call history on the mocks!